<debug> unittest 기반 tester 모듈 추가

ebbunnim opened the pull request:
건모님이 디버그 기능 이미 구현해주셨네요 ㅎㅎ 👍 
저는 reader말고 retrieval 부분들도 테스트 코드 만들어야 될 수도 있어서 모듈 형식으로 짜봤어요!
소모적인 방향이라면, PR 안받아도 됩니다!

ebbunnim committed:
<feat> Add unittest module

ebbunnim committed:
<fix> add wandb to gitignore

ebbunnim committed:
<fix> delete wandb folder

ebbunnim committed:
<fix> fix sampling

ebbunnim committed:
<fix> fix prepare_dataset

ggm1207 commented on tools.py:
```diff
@@ -51,6 +58,7 @@ def get_args():
     arg_parser.add_argument("--run_cnt", type=int, default=1)
     arg_parser.add_argument("--seeds", type=str2intlist, default=SEEDS)
     arg_parser.add_argument("--data_path", type=str, default="../input/")
+    arg_parser.add_argument("--debug", type=str2bool, default="False")
```
이 부분 default=False 로 바꾸시고 Merge 하시면 될 것 같습니다!! 
코드 보시느라 고생 많으셨습니다! :smile:

ggm1207 commented on tools.py:
```diff
@@ -26,6 +26,13 @@ def str2intlist(v):
+    elif v.lower() in ('false', 'f', 'no', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Debug 모드는 true/false 중 하나를 선택해주세요. ')
```
str2bool 타입이 다른 argument에서도 충분히 재활용 될 수 있는 함수라 raise 부분을 기존에 존재하던 것으로 바꿔도 될 것 같습니다!

ggm1207 commented on tester.py:
```diff
@@ -0,0 +1,73 @@
+                    compute_metrics=compute_metrics,
+                )
+
+                trainer.train()
```
헉, 단위 테스트를 하실 줄은 몰랐는데.. 대단하십니다. :+1:

그러면 단위 테스트는 용도가 명확하게 분리가 되니 run.py에 통합하는 것 보다 tester.py 에서 debug 명령 없이 실행하는 것이 좋을 것 같습니다!

```bash
python -m run --strategis ST01,ST02 --debug True  # Train 데이터 사이즈를 줄여서 빠르게 한 번 훑어보기 
python -m tester  # 각종 모듈들이 제대로 동작하는지 확인
```

추가로 README에 실행방법 작성해주시면 됩니다!

ggm1207 reviewed (commented):
방향성도 너무 좋고 코드도 잘 짜셨습니다!! debug랑 unittest가 좀 다른 역할을 가졌으면 해서 리뷰 남겨봤습니다. 이 방향으로 수정이 가능할까요?

ebbunnim committed:
<tmp> before merge

ebbunnim committed:
<fix> merge main

Merge branch 'main' of https://github.com/VumBleBot/odqa_baseline_code into ebbunnim

ebbunnim committed:
<fix> fix execute rule and remake sub functions

ebbunnim committed:
<docs> update readme(TDD)

ebbunnim committed:
<docs> update readme(TDD)
